### PR TITLE
refact(score): minor refact, doc & linting of the score package

### PR DESCRIPTION
**Pull request by fredbi**
* re-documented methods and differences between v1 and v2
    * fixed data race when calling the constructor
    * refactored debug code to be less intrusive

This PR  is a follow-up on #87, which introduced more test coverage to limit the risk of regression.
    
Signed-off-by: Frederic BIDON <fredbi@yahoo.com>

### CODE REVIEW

**Comment by fredbi** [comment]
cc: @matthyx @dwertent 

This PR stemmed from the desire to understand better how the risk score is computed.

I started by beefing up the tests, to get a better grasp of the data structures.

I could see a few itching issues (I will report a separate issue shortly):

* I don't understand why the ScoreUtil is a package-level singleton. I've fixed the data race, but it really feels like we should not do that.
* I've noticed a difference in the score computed with models V1 and V2. Simulating equivalent situations with both versions of our object model, weights do not follow the same rule. I am wondering if this was desired in the first place.
* Supporting packages with multiple versions is a pain. I don't know if V1 is here to stay for a while. If it is, I could dedicate a little extra time to refactoring how we should define structs with such variants. The current way tends to obfuscate the intent.

**Review by dwertent — APPROVED** [review]
Great improvement!
I will wait for #87 to merge before we merge this PR.

**Comment by fredbi** [comment]
@dwertent  rebased

**Pull request merged**
